test/loudness-analyser.cpp

```diff
@@ -6,72 +6,354 @@
+    // source_3341_14.setMomentaryLoudness( -19.0f );
+
+    essences.push_back( source_3341_14 );
+
+    FileConfiguration source_3341_15( "seq-3341-15-24bit.wav.wav" );
+    source_3341_15.setTruePeakInDbTP( -6.0f );
+    FileConfiguration source_3341_16( "seq-3341-16-24bit.wav.wav" );
+    source_3341_16.setTruePeakInDbTP( -6.0f );
+
+    FileConfiguration source_3341_17( "seq-3341-17-24bit.wav.wav" );
+    source_3341_17.setTruePeakInDbTP( -6.0f );
+    FileConfiguration source_3341_18( "seq-3341-18-24bit.wav.wav" );
+    source_3341_18.setTruePeakInDbTP( -6.0f );
+    FileConfiguration source_3341_19( "seq-3341-19-24bit.wav.wav" );
+    source_3341_19.setTruePeakInDbTP( +3.0f );
+    FileConfiguration source_3341_20( "seq-3341-20-24bit.wav.wav" );
+    source_3341_20.setTruePeakInDbTP( -0.0f );
+    FileConfiguration source_3341_21( "seq-3341-21-24bit.wav.wav" );
+    source_3341_21.setTruePeakInDbTP( -0.0f );
+    FileConfiguration source_3341_22( "seq-3341-22-24bit.wav.wav" );
+    source_3341_22.setTruePeakInDbTP( -0.0f );
+    FileConfiguration source_3341_23( "seq-3341-23-24bit.wav.wav" );
+    source_3341_23.setTruePeakInDbTP( -0.0f );
+
+    essences.push_back( source_3341_15 );
+    essences.push_back( source_3341_16 );
+    essences.push_back( source_3341_17 );
+    essences.push_back( source_3341_18 );
+    essences.push_back( source_3341_19 );
+    essences.push_back( source_3341_20 );
+    essences.push_back( source_3341_21 );
+    essences.push_back( source_3341_22 );
+    essences.push_back( source_3341_23 );
+
+
+    FileConfiguration source_3342_1( "seq-3342-1-16bit.wav" );
+    source_3342_1.setIntegratedRange( 10.0f );
+    FileConfiguration source_3342_2( "seq-3342-2-16bit.wav" );
+    source_3342_2.setIntegratedRange( 5.0f );
+    FileConfiguration source_3342_3( "seq-3342-3-16bit.wav" );
+    source_3342_3.setIntegratedRange( 20.0f );
+    FileConfiguration source_3342_4( "seq-3342-4-16bit.wav" );
+    source_3342_4.setIntegratedRange( 15.0f );
+
+    FileConfiguration source_3342_5( "seq-3341-7_seq-3342-5-24bit.wav" );
+    source_3342_5.setIntegratedRange( 5.0f );
+
+    essences.push_back( source_3342_1 );
+    essences.push_back( source_3342_2 );
+    essences.push_back( source_3342_3 );
+    essences.push_back( source_3342_4 );
+    essences.push_back( source_3342_5 );
+
+    return essences;
 }
 
-TEST(Case1kHzSine, Test40LUFS)
+INSTANTIATE_TEST_CASE_P(EbuTestEssences,
+                        CaseLoudnessAnalysis,
+                        ::testing::ValuesIn(getEbuEssences()));
+
+TEST_P(CaseLoudnessAnalysis, Test)
 {
-    checkEBUR128Analysis("1kHz Sine -40 LUFS-16bit.wav", -40, -40, -40);
+    if( ! isnan( _cfg.getIntegratedLoudness() ) )
+    {
+        ASSERT_NEAR( _loudness.getIntegratedLoudness(), _cfg.getIntegratedLoudness(), 0.1 );
+    }
+    if( ! isnan( _cfg.getIntegratedRange() ) )
+    {
+        ASSERT_NEAR( _loudness.getIntegratedRange(), _cfg.getIntegratedRange(), 1.0 );
```

Comment:
it's limits defined in EBU specifications : https://tech.ebu.ch/docs/tech/tech3342.pdf
LRA is valid when the system can found the value around +-1 LU around the (LRA = 10 ±1
LU)